### DISCUSSION TITLE
input onChange, onKeyUp, etc.

### DISCUSSION THREAD

**Discussion by alvaroponguta** [dowjones/react-dropdown-tree-select#492] [Q&A]
Hello everybody! Im looking for a way to use this events in the input, but im not finding any of those on the props. Or maybe if someone know a way to know when the user begins or finish typing will be awesome.

Thank you everyone.

**Comment by mrchief** [comment]
The `onChange` events are captured internally for filtering the tree. Bubbling them up will be disastrous as React will re-render every time (assuming there will be some state change by the event handler).
